tools/attach_logs_in_jira.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import logging
+import re
+import shutil
+import tarfile
+from pathlib import Path
+from functools import partial
+
+import jira
+
+from consts import JIRA_PROJECT, JIRA_SUMMARY, JIRA_SERVER
+
+JIRA_ATTACHMENT_MAX_SIZE = 50 * 2**20  # 50MB
+
+log = logging.getLogger(__name__)
+
+
+def _get_or_create_jira_ticket(jira_client: jira.JIRA, summary: str) -> jira.Issue:
+    matching_issues = jira_client.search_issues(f'project = "{JIRA_PROJECT}" AND summary = {summary}')
+
+    if len(matching_issues) == 0:
+        log.info("Creating a placeholder issue for %s", summary)
+        return jira_client.create_issue(
+            project=JIRA_PROJECT,
+            summary=summary,
+            description=(
+                "Logs are still being uploaded.\nTicket description and signatures will get processed shortly..."
+            ),
+        )
+
+    if len(matching_issues) == 1:
+        log.debug("Using pre-existing issue %s", matching_issues[0])
+        return matching_issues[0]
+
+    raise RuntimeError(f"Found more than one matching issue for {cluster_logs}: {' '.join(matching_issues)}")
+
+
+def _archive_dir(directory: Path) -> Path:
+    archived_file = Path(f"{directory}.tar.gz")
```

Comment:
nit: with_suffix can save you formatting and the `Path` cast

```suggestion
    archived_file = directory.with_suffix(".tar.gz")
```